### PR TITLE
ci: Run self-hosted ci

### PR DESCRIPTION
**Pull request by maflcko**
Due to our heavy use of the Cirrus CI community cluster, some tasks may take a long time to get scheduled. While it is possible to use "Compute Credits" to get immediately scheduled on the cluster, I couldn't find a sponsor that'd be willing to cover the total cost, if all tasks were paid for with credits.

However, it is also possible to bring our own runners to Cirrus CI.

For testing purposes, a single task will be transformed to run on the DrahtBot infrastructure. If all goes well, the other tasks can be moved, too.

### CODE REVIEW

**Comment by jarolrod** [comment]
Concept ACK

DrahtBot strong 🦾

File: .cirrus.yml [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
   TEST_RUNNER_PORT_MIN: "14000"  # Must be larger than 12321, which is used for the http cache. See https://cirrus-ci.org/guide/writing-tasks/#http-cache
   CCACHE_SIZE: "200M"
   CCACHE_DIR: "/tmp/ccache_dir"
+  CCACHE_NOHASHDIR: "1"  # Debug info might contain a stale path if the build dir changes, but this is fine
+
+cirrus_ephemeral_worker_template_env: &CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
+  DANGER_RUN_CI_ON_HOST: "1"  # Containers willl be discarded after the run, so there is no risk that the ci scripts modify the system
+
+persistent_worker_template_env: &PERSISTENT_WORKER_TEMPLATE_ENV
+  CI_RESTART_DOCKER_BEFORE_RUN: "1"
```
Comment (hebasto) [inline .cirrus.yml:15]: I'd leave `CI_`-prefixed names for CI :)
Maybe just `RESTART_DOCKER_BEFORE_RUN`?

File: .cirrus.yml [hunk 2]
```diff
@@ -3,10 +3,19 @@
...
   PACKAGE_MANAGER_INSTALL: "apt-get update && apt-get install -y"
   MAKEJOBS: "-j4"
-  DANGER_RUN_CI_ON_HOST: "1"  # Containers will be discarded after the run, so there is no risk that the ci scripts modify the system
   TEST_RUNNER_PORT_MIN: "14000"  # Must be larger than 12321, which is used for the http cache. See https://cirrus-ci.org/guide/writing-tasks/#http-cache
   CCACHE_SIZE: "200M"
   CCACHE_DIR: "/tmp/ccache_dir"
+  CCACHE_NOHASHDIR: "1"  # Debug info might contain a stale path if the build dir changes, but this is fine
+
+cirrus_ephemeral_worker_template_env: &CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
+  DANGER_RUN_CI_ON_HOST: "1"  # Containers willl be discarded after the run, so there is no risk that the ci scripts modify the system
```
Comment (hebasto) [inline .cirrus.yml:12]: typo:
```suggestion
  DANGER_RUN_CI_ON_HOST: "1"  # Containers will be discarded after the run, so there is no risk that the ci scripts modify the system
```

**Review by hebasto — COMMENTED** [review]
Concept ACK.

Commit _"ci: Restart docker before run"_. Why we need to restart docker? Isn't it a CI's responsibility to create a new container for each task?

File: .cirrus.yml (hunk 2 above)
Comment (MarcoFalke) [reply .cirrus.yml]: thanks, fixed

File: .cirrus.yml (hunk 1 above)
Comment (MarcoFalke) [reply .cirrus.yml]: Renamed to `RESTART_CI_DOCKER_BEFORE_RUN`. I think `ci` is fine in the name as long as it is not a prefix

**Comment by MarcoFalke** [comment]
> Why we need to restart docker? Isn't it a CI's responsibility to create a new container for each task?

This can simplify the local workflow when ci tasks are executed repeatedly. Also, it is needed for persistent runners, which are not ephemeral. See also https://cirrus-ci.org/guide/persistent-workers/#what-is-a-persistent-worker

**Review by ytbigvinny27 — APPROVED** [review]
---🔜🤌🏽🤌🏽🔝

**Review by hebasto — APPROVED** [review]
ACK fa41a917356a7f5e0f3286b227ced7f2e6797e67, I have reviewed the code and it looks OK.

CI [log](https://cirrus-ci.com/task/5664154569670656) looks OK.

**Comment by hebasto** [comment]
Unrelated to this PR, but why is `depends_release` cache populated _after_ CI run?

Screenshot from 2021-04-10 16-04-35

**Comment by MarcoFalke** [comment]
> Unrelated ...

Fixed in #21653

**Review by RonSherfey — COMMENTED** [review]
reviewed changes

**Review by 78051301012 — COMMENTED** [review]
.

**Pull request merged**